hardware/detect.py

```diff
@@ -689,7 +689,28 @@ def find_element(xml, xml_spec, sys_subtype,
     return True
 
 
+def _from_file(fname, mapping=None, default=None):
+    file_exists = os.path.exists(fname)
+    value = None
```

Comment:
Shouldn't it be `value = default`?

Reply:
It can be but that expression will only get returned when file_exists == False, and when that happens the the 'value' shoudl be ignored.

I don't really like the function returning a tuple.  Perhaps this confusion is 'worse' than the duplication I was trying to remove?